core/src/main/scala/org/bitcoins/core/protocol/script/ScriptWitness.scala

```diff
@@ -106,6 +110,21 @@ sealed abstract class P2WSHWitnessV0 extends ScriptWitnessV0 {
     relevantStack.map(ECDigitalSignature.fromBytes)
   }
 
+  // Note that this is not guaranteed to work for non-standard script signatures
+  lazy val scriptSignature: ScriptSignature = {
```

Comment:
Should this be an `Option[ScriptSignature]`? 

This throws on the case where the `stack` is empty (`.tail`)

Reply:
Upon further evaluation, I believe that this actually does work in all cases, and the thing that doesn't work is the apply method below which takes in a ScriptSignature (which will fail to act as expected if there's anything other than push ops). So no `Option`, but you're right that we should `require` that there be a redeem script (i.e. `stack.nonEmpty`) at the top of this trait